src/shell.js

```diff
@@ -38,6 +38,13 @@ var Module = typeof {{{ EXPORT_NAME }}} !== 'undefined' ? {{{ EXPORT_NAME }}} :
 #endif // USE_CLOSURE_COMPILER
 #endif // SIDE_MODULE
 
+#if ((MAYBE_WASM2JS && WASM != 2) || MODULARIZE) && (MIN_CHROME_VERSION < 33 || MIN_EDGE_VERSION < 12 || MIN_FIREFOX_VERSION < 29 || MIN_IE_VERSION != TARGET_NOT_SUPPORTED || MIN_SAFARI_VERSION < 80000) // https://caniuse.com/#feat=promises
+// Include a Promise polyfill for legacy browsers. This is needed either for
+// wasm2js, where we polyfill the wasm API which needs Promises, or when using
+// modularize which creates a Promise for when the module is ready.
+#include "promise_polyfill.js"
```

Comment:
What about shell_minimal?

Reply:
Not sure it's worth supporting every flag combination there. I think it's reasonable to say that if someone wants legacy VM support, which bundles a lot of polyfill code anyhow, that they should use the normal runtime and not the minimal one.